In [1]:
import sys
import glob
 
sys.path.insert(0, '/Users/rekerl/Documents/Programs/gpc/')
 
from gpc import *
from gpc import compare
from gpc import compare_func as CF

# Full Test of Genotype/Phenotype Correlation Tool

### Upload test patients

In [2]:
allPatients = AllPatients('../phenopackets/cohort/*.json')

### Pull out all diseases with too few patients for testing purposes

In [3]:
byDisease = allPatients.split_by_disease()

In [4]:
too_small = []
for d, dis in byDisease.items():
    if dis.count_patients < 10:
        too_small.append(d)
        
for i in too_small:
    byDisease.pop(i)
print(byDisease.keys())

dict_keys(['OMIM:617425', 'OMIM:616900', 'OMIM:616907', 'OMIM:616975', 'OMIM:159550'])


### Run basic Missense mutation vs Nonsense mutation

In [5]:
for d, dis in byDisease.items():
    result = compare.RunStats(dis, CF.is_missense, CF.is_nonsense)
    print(result)

Total hpo terms: 171


INFO:pyensembl.sequence_data:Loaded sequence dictionary from /Users/rekerl/Library/Caches/pyensembl/GRCh37/ensembl75/Homo_sapiens.GRCh37.75.cdna.all.fa.gz.pickle
INFO:pyensembl.sequence_data:Loaded sequence dictionary from /Users/rekerl/Library/Caches/pyensembl/GRCh37/ensembl75/Homo_sapiens.GRCh37.75.ncrna.fa.gz.pickle
INFO:pyensembl.sequence_data:Loaded sequence dictionary from /Users/rekerl/Library/Caches/pyensembl/GRCh37/ensembl75/Homo_sapiens.GRCh37.75.pep.all.fa.gz.pickle
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: Performance

counts                                       1 w/ hpo  1 w/o hpo  2 w/ hpo  \
HP:0000158-Macroglossia                           1.0        9.0       0.0   
HP:0001943-Hypoglycemia                           1.0        9.0       0.0   
HP:0002205-Recurrent respiratory infections       1.0        9.0       0.0   
HP:0001655-Patent foramen ovale                   1.0        9.0       0.0   
HP:0001888-Lymphopenia                            1.0        9.0       0.0   
...                                               ...        ...       ...   
HP:0008445-Cervical spinal canal stenosis         1.0        9.0       0.0   
HP:0008947-Infantile muscular hypotonia           1.0        9.0       0.0   
HP:0001508-Failure to thrive                      3.0        7.0       0.0   
HP:0001263-Global developmental delay             3.0        7.0       0.0   
HP:0001880-Eosinophilia                           2.0        8.0       0.0   

counts                                       2 w/o hpo  pval  


/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fr

/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fr

counts                                  1 w/ hpo  1 w/o hpo  2 w/ hpo  \
HP:0002375-Hypokinesia                       0.0        6.0       0.0   
HP:0001935-Microcytic anemia                 0.0        6.0       0.0   
HP:0005667-Os odontoideum                    0.0        6.0       0.0   
HP:0001272-Cerebellar atrophy                0.0        6.0       0.0   
HP:0002280-Enlarged cisterna magna           0.0        6.0       0.0   
...                                          ...        ...       ...   
HP:0001249-Intellectual disability           5.0        1.0       2.0   
HP:0000582-Upslanted palpebral fissure       1.0        5.0       0.0   
HP:0001562-Oligohydramnios                   1.0        5.0       0.0   
HP:0012444-Brain atrophy                     1.0        5.0       0.0   
HP:0000414-Bulbous nose                      2.0        4.0       0.0   

counts                                  2 w/o hpo      pval  
HP:0002375-Hypokinesia                        7.0  1.000000  

/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fr

/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fr

/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fr

/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fr

/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fr

counts                                        1 w/ hpo  1 w/o hpo  2 w/ hpo  \
HP:0002616-Aortic root aneurysm                    1.0        9.0       0.0   
HP:0002058-Myopathic facies                        1.0        9.0       0.0   
HP:0003198-Myopathy                                1.0        9.0       0.0   
HP:0030842-Choking episodes                        1.0        9.0       0.0   
HP:0000407-Sensorineural hearing impairment        1.0        9.0       0.0   
...                                                ...        ...       ...   
HP:0002020-Gastroesophageal reflux                 3.0        7.0       0.0   
HP:0011480-Unilateral microphthalmos               1.0        9.0       0.0   
HP:0001260-Dysarthria                              1.0        9.0       0.0   
HP:0006808-Cerebral hypomyelination                5.0        5.0       0.0   
HP:0002079-Hypoplasia of the corpus callosum       4.0        6.0       0.0   

counts                                        2 w/o

### Pull out all patients with disease OMIM:616907, print Patient ID and the top effect on the protein

In [6]:
patswith = byDisease['OMIM:616900']
for p, pat in patswith.all_patients.items():
    print(p)
    print(pat.variant.top_effect.short_description)
    

PMID:27040691-Bhoj-2016-TBCK-1-1
splice-donor
PMID:30103036-Zapata-Aldana-2019-TBCK-Patient_2
p.252fs*
PMID:27040691-Bhoj-2016-TBCK-4-2
splice-acceptor
PMID:27275012-Guerreiro-2016-TBCK-II-4
p.M268fs
PMID:27040692-Chong-2016-TBCK-D-II-1
p.R126*
PMID:27040691-Bhoj-2016-TBCK-6-2
p.N457fs
PMID:27040691-Bhoj-2016-TBCK-3-1
p.L551P
PMID:27040692-Chong-2016-TBCK-C-II-1
p.R511H
PMID:27040691-Bhoj-2016-TBCK-5-1
p.R126*
PMID:27040692-Chong-2016-TBCK-B-IV-4
p.K455*
PMID:27275012-Guerreiro-2016-TBCK-II-3
p.M268fs
PMID:27040691-Bhoj-2016-TBCK-2-1
p.R126*
PMID:27040692-Chong-2016-TBCK-B-IV-6
p.K455*
PMID:27040691-Bhoj-2016-TBCK-6-1
p.N457fs
PMID:27040692-Chong-2016-TBCK-A-II-1
p.R126*
PMID:27040691-Bhoj-2016-TBCK-1-2
splice-donor
PMID:27040691-Bhoj-2016-TBCK-8-1
p.R126*
PMID:27040691-Bhoj-2016-TBCK-4-1
splice-acceptor
PMID:30103036-Zapata-Aldana-2019-TBCK-Patient_1
p.252fs*


### Grabbed an example patient to look at protein features

In [7]:
test = patswith.all_patients.get('PMID:27040691-Bhoj-2016-TBCK-1-2')

test.variant.protein.features

{'Chain': [{'start': 1, 'end': 893}],
 'Domain': [{'start': 1, 'end': 273},
  {'start': 466, 'end': 651},
  {'start': 790, 'end': 889}],
 'Region': [{'start': 710, 'end': 749}],
 'Compositional bias': [{'start': 710, 'end': 730}],
 'Alternative sequence': [{'start': 89, 'end': 152},
  {'start': 161, 'end': 199}],
 'Natural variant': [{'start': 66, 'end': 66},
  {'start': 151, 'end': 151},
  {'start': 265, 'end': 265},
  {'start': 266, 'end': 266},
  {'start': 425, 'end': 425},
  {'start': 471, 'end': 471},
  {'start': 489, 'end': 489},
  {'start': 503, 'end': 503},
  {'start': 511, 'end': 511},
  {'start': 551, 'end': 551},
  {'start': 692, 'end': 692},
  {'start': 806, 'end': 806}],
 'Sequence conflict': [{'start': 601, 'end': 601}]}

### Run a test that compares HPO terms vs the variant being in one or the other protein domain

In [8]:
result = compare.RunStats(patswith, CF.in_domain, CF.in_domain, [1, 273], [466, 651])
print(result.sort_values(by=['pval']))

Total hpo terms: 130


/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107092429T>C with effect splice-acceptor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/g

/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107092429T>C with effect splice-acceptor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/g

/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107092429T>C with effect splice-acceptor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107092429T>C with effect splice-acceptor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Program

/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107092429T>C with effect splice-acceptor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107092429T>C with effect splice-acceptor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Program

/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107092429T>C with effect splice-acceptor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/g

/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107092429T>C with effect splice-acceptor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/g

/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107092429T>C with effect splice-acceptor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/g

/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107092429T>C with effect splice-acceptor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping varia

/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107092429T>C with effect splice-acceptor
  warnings.warn('This effect has no location, skipping varia

counts                                      1 w/ hpo  1 w/o hpo  2 w/ hpo  \
HP:0000878-11 pairs of ribs                      0.0        9.0       1.0   
HP:0000717-Autism                                0.0        9.0       1.0   
HP:0001500-Broad finger                          0.0        9.0       1.0   
HP:0011342-Mild global developmental delay       0.0        9.0       1.0   
HP:0007302-Bipolar affective disorder            0.0        9.0       1.0   
...                                              ...        ...       ...   
HP:0006380-Knee flexion contracture              1.0        8.0       0.0   
HP:0000384-Preauricular skin tag                 1.0        8.0       0.0   
HP:0001239-Wrist flexion contracture             1.0        8.0       0.0   
HP:0000664-Synophrys                             1.0        8.0       0.0   
HP:0000414-Bulbous nose                          0.0        9.0       0.0   

counts                                      2 w/o hpo      pval  
HP:000087

/Users/rekerl/Documents/Programs/gpc/gpc/compare.py:22: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  results.insert(1, hpo_id[0] + '-'+hpo_id[1], [var1_with_hpo, var1_without_hpo, var2_with_hpo, var2_without_hpo, p])
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107115874C>T with effect splice-donor
  warnings.warn('This effect has no location, skipping variant ' + pat.variant.variant_string + ' with effect ' + pat.variant.top_effect.short_description)
/Users/rekerl/Documents/Programs/gpc/gpc/compare_func.py:82: UserWarning: This effect has no location, skipping variant chr4 g.107092429T>C with effect splice-acceptor
  warnings.warn('This effect has no location, skipping varia